2022-03-11 19:57:34 [INFO ]  ################################################################################
2022-03-11 19:57:34 [INFO ]  
2022-03-11 19:57:34 [INFO ]  Installing FREEDOM ROBOTICS Micro Service
2022-03-11 19:57:34 [INFO ]  
2022-03-11 19:57:34 [INFO ]  Copyright Freedom Robotics Inc 2021. All rights reserved. Confidential.
2022-03-11 19:57:34 [INFO ]  
2022-03-11 19:57:34 [INFO ]  ################################################################################
2022-03-11 19:57:34 [DEBUG]  		EXECUTING SYSTEM COMMAND: /bin/bash -c "whoami"
2022-03-11 19:57:37 [DEBUG]  		SUCCESS RESPONSE CODE: 0, EXECUTION TIME: 3.27s
2022-03-11 19:57:37 [DEBUG]  		EXECUTING SYSTEM COMMAND: /bin/bash -c "sudo -nv"
2022-03-11 19:57:40 [DEBUG]  		STANDARD ERROR:
2022-03-11 19:57:40 [DEBUG]  			sudo: a password is required
2022-03-11 19:57:40 [DEBUG]  		ERROR RESPONSE CODE: 1, EXECUTION TIME: 3.30s
2022-03-11 19:57:40 [INFO ]  
2022-03-11 19:57:40 [INFO ]  Please enter the sudo password to enable install to set up the
Freedom SYSTEM SERVICE, Remote SSH, etc... for the Freedom Agent.
(This will time out after 60 seconds and continue without installing
the services.)
2022-03-11 19:57:40 [INFO ]  
2022-03-11 19:57:40 [DEBUG]  		EXECUTING SYSTEM COMMAND: /bin/bash -c "sudo -v"
2022-03-11 19:57:52 [DEBUG]  		SUCCESS RESPONSE CODE: 0, EXECUTION TIME: 11.64s
2022-03-11 19:57:54 [DEBUG]  		EXECUTING SYSTEM COMMAND: /bin/bash -c "sudo -nv"
2022-03-11 19:57:57 [DEBUG]  		SUCCESS RESPONSE CODE: 0, EXECUTION TIME: 3.27s
2022-03-11 19:57:57 [INFO ]  
2022-03-11 19:57:57 [INFO ]  ################################################################################
2022-03-11 19:57:57 [INFO ]  
2022-03-11 19:57:57 [INFO ]  Step 1/7: VERIFYING PREREQUISITES
2022-03-11 19:57:57 [DEBUG]  		EXECUTING SYSTEM COMMAND: /bin/bash -c "/usr/bin/python -c 'import sys
import os
if (hasattr(sys, \"base_prefix\") and sys.base_prefix != sys.prefix) or hasattr(sys, \"real_prefix\") or os.environ.get(\"CONDA_PREFIX\"):
    print(\"Detected we are dealing with a virtual environment for /usr/bin/python\")'"
2022-03-11 19:58:01 [DEBUG]  		SUCCESS RESPONSE CODE: 0, EXECUTION TIME: 3.76s
2022-03-11 19:58:01 [DEBUG]  		EXECUTING SYSTEM COMMAND: /bin/bash -c "/usr/bin/python -m pip install -qq --user distro"
2022-03-11 19:58:04 [DEBUG]  		STANDARD ERROR:
2022-03-11 19:58:04 [DEBUG]  			/usr/bin/python: No module named pip
2022-03-11 19:58:04 [DEBUG]  		ERROR RESPONSE CODE: 1, EXECUTION TIME: 3.30s
2022-03-11 19:58:04 [DEBUG]  		EXECUTING SYSTEM COMMAND: /bin/bash -c "/usr/bin/python -m pip &> /dev/null"
2022-03-11 19:58:08 [DEBUG]  		ERROR RESPONSE CODE: 1, EXECUTION TIME: 3.30s
2022-03-11 19:58:08 [INFO ]  Installing pip for python2.7: /usr/bin/python. Please stand by.
2022-03-11 19:58:08 [DEBUG]  		EXECUTING SYSTEM COMMAND: /bin/bash -c "sudo -n apt-get update"
2022-03-11 19:59:53 [DEBUG]  		SUCCESS RESPONSE CODE: 0, EXECUTION TIME: 105.73s
2022-03-11 19:59:53 [DEBUG]  		EXECUTING SYSTEM COMMAND: /bin/bash -c "/usr/bin/python -m pip install -qq --user distro"
2022-03-11 19:59:58 [DEBUG]  		STANDARD ERROR:
2022-03-11 19:59:58 [DEBUG]  			/usr/bin/python: No module named pip
2022-03-11 19:59:58 [DEBUG]  		ERROR RESPONSE CODE: 1, EXECUTION TIME: 4.54s
2022-03-11 19:59:58 [DEBUG]  		EXECUTING SYSTEM COMMAND: /bin/bash -c "sudo -n  curl https://bootstrap.pypa.io/pip/2.7/get-pip.py -o /tmp/get-pip.py && sudo -n  /usr/bin/python /tmp/get-pip.py --user"
2022-03-11 20:00:21 [DEBUG]  		STANDARD ERROR:
2022-03-11 20:00:21 [DEBUG]  			% Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
2022-03-11 20:00:21 [DEBUG]  			Dload  Upload   Total   Spent    Left  Speed
2022-03-11 20:00:21 [DEBUG]  			
2022-03-11 20:00:21 [DEBUG]  			0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
2022-03-11 20:00:21 [DEBUG]  			100 1863k  100 1863k    0     0  5664k      0 --:--:-- --:--:-- --:--:-- 5664k
2022-03-11 20:00:21 [DEBUG]  			DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. pip 21.0 will drop support for Python 2.7 in January 2021. More details about Python 2 support in pip can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support pip 21.0 will remove support for this functionality.
2022-03-11 20:00:21 [DEBUG]  			WARNING: The scripts pip, pip2 and pip2.7 are installed in '/root/.local/bin' which is not on PATH.
2022-03-11 20:00:21 [DEBUG]  			Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
2022-03-11 20:00:21 [DEBUG]  			WARNING: The script wheel is installed in '/root/.local/bin' which is not on PATH.
2022-03-11 20:00:21 [DEBUG]  			Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
2022-03-11 20:00:21 [DEBUG]  		SUCCESS RESPONSE CODE: 0, EXECUTION TIME: 23.03s
2022-03-11 20:00:21 [DEBUG]  		EXECUTING SYSTEM COMMAND: /bin/bash -c "which python3"
2022-03-11 20:00:24 [DEBUG]  		SUCCESS RESPONSE CODE: 0, EXECUTION TIME: 3.34s
2022-03-11 20:00:24 [DEBUG]  		EXECUTING SYSTEM COMMAND: /bin/bash -c "[running ~/.bashrc]
[running common_alias.bash]
/usr/bin/python3 -c 'import sys; print(\"{}.{}\".format(sys.version_info.major, sys.version_info.minor))'"
2022-03-11 20:00:28 [DEBUG]  		STANDARD ERROR:
2022-03-11 20:00:28 [DEBUG]  			/bin/bash: [running: command not found
2022-03-11 20:00:28 [DEBUG]  			/bin/bash: line 1: [running: command not found
2022-03-11 20:00:28 [DEBUG]  		SUCCESS RESPONSE CODE: 0, EXECUTION TIME: 3.56s
2022-03-11 20:00:28 [DEBUG]  Using the default python3 binary: [running ~/.bashrc]
[running common_alias.bash]
/usr/bin/python3
2022-03-11 20:00:28 [DEBUG]  		EXECUTING SYSTEM COMMAND: /bin/bash -c "[running ~/.bashrc]
[running common_alias.bash]
/usr/bin/python3 -c 'import sys; print(\"{}.{}\".format(sys.version_info.major, sys.version_info.minor))'"
2022-03-11 20:00:33 [DEBUG]  		STANDARD ERROR:
2022-03-11 20:00:33 [DEBUG]  			/bin/bash: [running: command not found
2022-03-11 20:00:33 [DEBUG]  			/bin/bash: line 1: [running: command not found
2022-03-11 20:00:33 [DEBUG]  		SUCCESS RESPONSE CODE: 0, EXECUTION TIME: 4.87s
2022-03-11 20:00:33 [DEBUG]  		EXECUTING SYSTEM COMMAND: /bin/bash -c "[running ~/.bashrc]
[running common_alias.bash]
/usr/bin/python3 -m pip &> /dev/null"
2022-03-11 20:00:43 [DEBUG]  		STANDARD ERROR:
2022-03-11 20:00:43 [DEBUG]  			/bin/bash: [running: command not found
2022-03-11 20:00:43 [DEBUG]  			/bin/bash: line 1: [running: command not found
2022-03-11 20:00:43 [DEBUG]  		SUCCESS RESPONSE CODE: 0, EXECUTION TIME: 9.94s
2022-03-11 20:00:44 [DEBUG]  Urllib2 can be use for ssl communication.
2022-03-11 20:00:44 [DEBUG]  		EXECUTING SYSTEM COMMAND: /bin/bash -c "/usr/bin/python -c 'import sys; print(\"{}.{}\".format(sys.version_info.major, sys.version_info.minor))'"
2022-03-11 20:00:47 [DEBUG]  		SUCCESS RESPONSE CODE: 0, EXECUTION TIME: 3.33s
2022-03-11 20:00:47 [DEBUG]  		EXECUTING SYSTEM COMMAND: /bin/bash -c "which ros2"
2022-03-11 20:00:50 [DEBUG]  		ERROR RESPONSE CODE: 1, EXECUTION TIME: 3.36s
2022-03-11 20:00:50 [DEBUG]  Executable `ros2` was not found using current PATH: Command '/bin/bash -c "which ros2"' returned non-zero exit status 1
2022-03-11 20:00:50 [DEBUG]  		EXECUTING SYSTEM COMMAND: /bin/bash -c "/usr/bin/python -m pip install  --user  rospkg"
2022-03-11 20:00:54 [DEBUG]  		STANDARD ERROR:
2022-03-11 20:00:54 [DEBUG]  			/usr/bin/python: No module named pip
2022-03-11 20:00:54 [DEBUG]  		ERROR RESPONSE CODE: 1, EXECUTION TIME: 3.43s
2022-03-11 20:00:54 [INFO ]  
2022-03-11 20:00:54 [INFO ]  
2022-03-11 20:00:54 [INFO ]  
2022-03-11 20:00:54 [CRITI]  [31m################################################################################[39m
2022-03-11 20:00:54 [CRITI]  [31m[39m
2022-03-11 20:00:54 [CRITI]  [31mINSTALL DID NOT COMPLETE SUCCESSFULLY[39m
2022-03-11 20:00:54 [CRITI]  [31m[39m
2022-03-11 20:00:54 [CRITI]  [31mFailed to install dependent ros python packages: [running ~/.bashrc]
[running common_alias.bash][39m
2022-03-11 20:00:54 [CRITI]  [31m[39m
2022-03-11 20:00:54 [CRITI]  [31m############################################################[39m
2022-03-11 20:00:54 [CRITI]  [31m[39m
2022-03-11 20:00:54 [CRITI]  [31m( NO STACK TRACE REPORTED )[39m
2022-03-11 20:00:54 [CRITI]  [31m[39m
2022-03-11 20:00:54 [CRITI]  [31m############################################################[39m
2022-03-11 20:00:54 [CRITI]  [31m[39m
2022-03-11 20:00:54 [CRITI]  [31mSTOPPING AGENT IF NECESSARY AND UNINSTALLING[39m
2022-03-11 20:00:54 [DEBUG]  		EXECUTING SYSTEM COMMAND: /bin/bash -c "sudo -n service freedomrobotics stop"
2022-03-11 20:01:10 [DEBUG]  		STANDARD ERROR:
2022-03-11 20:01:10 [DEBUG]  			Failed to stop freedomrobotics.service: Unit freedomrobotics.service not loaded.
2022-03-11 20:01:10 [DEBUG]  		ERROR RESPONSE CODE: 5, EXECUTION TIME: 16.31s
2022-03-11 20:01:10 [CRITI]  [31m    - AGENT NOT RUNNING[39m
2022-03-11 20:01:10 [DEBUG]  		EXECUTING SYSTEM COMMAND: /bin/bash -c "sudo -n systemctl daemon-reload"
2022-03-11 20:01:16 [DEBUG]  		SUCCESS RESPONSE CODE: 0, EXECUTION TIME: 5.99s
2022-03-11 20:01:16 [DEBUG]  Systemd / Systemctl are running so registering as systemd service
2022-03-11 20:01:16 [DEBUG]  		EXECUTING SYSTEM COMMAND: /bin/bash -c "/usr/bin/python -m pip show freedomrobotics | grep \"Location: \""
2022-03-11 20:01:20 [DEBUG]  		STANDARD ERROR:
2022-03-11 20:01:20 [DEBUG]  			/usr/bin/python: No module named pip
2022-03-11 20:01:20 [DEBUG]  		ERROR RESPONSE CODE: 1, EXECUTION TIME: 3.58s
2022-03-11 20:01:20 [DEBUG]  Unhandled exception trying to remove service: Could not find freedomrobotics python package install location:  Command '/bin/bash -c "/usr/bin/python -m pip show freedomrobotics | grep \"Location: \""' returned non-zero exit status 1
2022-03-11 20:01:20 [DEBUG]  		EXECUTING SYSTEM COMMAND: /bin/bash -c "/usr/bin/python -c 'import sys; print(\"{}.{}\".format(sys.version_info.major, sys.version_info.minor))'"
2022-03-11 20:01:23 [DEBUG]  		SUCCESS RESPONSE CODE: 0, EXECUTION TIME: 3.50s
2022-03-11 20:01:23 [DEBUG]  		EXECUTING SYSTEM COMMAND: /bin/bash -c "/usr/bin/python -m pip show freedomrobotics"
2022-03-11 20:01:27 [DEBUG]  		STANDARD ERROR:
2022-03-11 20:01:27 [DEBUG]  			/usr/bin/python: No module named pip
2022-03-11 20:01:27 [DEBUG]  		ERROR RESPONSE CODE: 1, EXECUTION TIME: 3.49s
2022-03-11 20:01:27 [CRITI]  [31m    - PACKAGE NOT INSTALLED ([running ~/.bashrc]
[running common_alias.bash]
2.7)[39m
2022-03-11 20:01:27 [DEBUG]  		EXECUTING SYSTEM COMMAND: /bin/bash -c "[running ~/.bashrc]
[running common_alias.bash]
/usr/bin/python3 -c 'import sys; print(\"{}.{}\".format(sys.version_info.major, sys.version_info.minor))'"
2022-03-11 20:01:30 [DEBUG]  		STANDARD ERROR:
2022-03-11 20:01:30 [DEBUG]  			/bin/bash: [running: command not found
2022-03-11 20:01:30 [DEBUG]  			/bin/bash: line 1: [running: command not found
2022-03-11 20:01:30 [DEBUG]  		SUCCESS RESPONSE CODE: 0, EXECUTION TIME: 3.65s
2022-03-11 20:01:30 [DEBUG]  		EXECUTING SYSTEM COMMAND: /bin/bash -c "[running ~/.bashrc]
[running common_alias.bash]
/usr/bin/python3 -m pip show freedomrobotics"
2022-03-11 20:01:40 [DEBUG]  		STANDARD ERROR:
2022-03-11 20:01:40 [DEBUG]  			/bin/bash: [running: command not found
2022-03-11 20:01:40 [DEBUG]  			/bin/bash: line 1: [running: command not found
2022-03-11 20:01:40 [DEBUG]  			WARNING: Package(s) not found: freedomrobotics
2022-03-11 20:01:40 [DEBUG]  		ERROR RESPONSE CODE: 1, EXECUTION TIME: 9.42s
2022-03-11 20:01:40 [CRITI]  [31m    - PACKAGE NOT INSTALLED ([running ~/.bashrc]
[running common_alias.bash]
3.8)[39m
2022-03-11 20:01:40 [CRITI]  [31m[39m
2022-03-11 20:01:40 [CRITI]  [31mUPLOADING INSTALL ERROR REPORT[39m
2022-03-11 20:01:40 [DEBUG]  		EXECUTING SYSTEM COMMAND: /bin/bash -c "pgrep fr_agent"
2022-03-11 20:01:44 [DEBUG]  		ERROR RESPONSE CODE: 1, EXECUTION TIME: 3.96s
2022-03-11 20:01:45 [DEBUG]  		EXECUTING SYSTEM COMMAND: /bin/bash -c "pgrep fr_agent"
2022-03-11 20:01:48 [DEBUG]  		ERROR RESPONSE CODE: 1, EXECUTION TIME: 3.80s
2022-03-11 20:01:49 [DEBUG]  		EXECUTING SYSTEM COMMAND: /bin/bash -c "pgrep fr_agent"
2022-03-11 20:01:53 [DEBUG]  		ERROR RESPONSE CODE: 1, EXECUTION TIME: 3.74s
2022-03-11 20:01:54 [DEBUG]  		EXECUTING SYSTEM COMMAND: /bin/bash -c "pgrep fr_agent"
2022-03-11 20:01:58 [DEBUG]  		ERROR RESPONSE CODE: 1, EXECUTION TIME: 3.73s
2022-03-11 20:01:59 [DEBUG]  		EXECUTING SYSTEM COMMAND: /bin/bash -c "pgrep fr_agent"
2022-03-11 20:02:03 [DEBUG]  		ERROR RESPONSE CODE: 1, EXECUTION TIME: 3.75s
2022-03-11 20:02:04 [DEBUG]  		EXECUTING SYSTEM COMMAND: /bin/bash -c "service_path=\$(which service) && output=\$( \$service_path ssh status ) && echo \"\$output\""
2022-03-11 20:02:11 [DEBUG]  		SUCCESS RESPONSE CODE: 0, EXECUTION TIME: 7.48s
2022-03-11 20:02:11 [DEBUG]  		EXECUTING SYSTEM COMMAND: /bin/bash -c "/usr/bin/python -m pip show freedomrobotics"
2022-03-11 20:02:14 [DEBUG]  		STANDARD ERROR:
2022-03-11 20:02:14 [DEBUG]  			/usr/bin/python: No module named pip
2022-03-11 20:02:14 [DEBUG]  		ERROR RESPONSE CODE: 1, EXECUTION TIME: 3.35s
2022-03-11 20:02:14 [DEBUG]  		EXECUTING SYSTEM COMMAND: /bin/bash -c "/usr/bin/python -c 'import sys; print(\"{}.{}\".format(sys.version_info.major, sys.version_info.minor))'"
2022-03-11 20:02:18 [DEBUG]  		SUCCESS RESPONSE CODE: 0, EXECUTION TIME: 3.44s
2022-03-11 20:02:18 [DEBUG]  		EXECUTING SYSTEM COMMAND: /bin/bash -c "[running ~/.bashrc]
[running common_alias.bash]
/usr/bin/python3 -m pip show freedomrobotics"
2022-03-11 20:02:27 [DEBUG]  		STANDARD ERROR:
2022-03-11 20:02:27 [DEBUG]  			/bin/bash: [running: command not found
2022-03-11 20:02:27 [DEBUG]  			/bin/bash: line 1: [running: command not found
2022-03-11 20:02:27 [DEBUG]  			WARNING: Package(s) not found: freedomrobotics
2022-03-11 20:02:27 [DEBUG]  		ERROR RESPONSE CODE: 1, EXECUTION TIME: 9.49s
2022-03-11 20:02:27 [DEBUG]  		EXECUTING SYSTEM COMMAND: /bin/bash -c "[running ~/.bashrc]
[running common_alias.bash]
/usr/bin/python3 -c 'import sys; print(\"{}.{}\".format(sys.version_info.major, sys.version_info.minor))'"
2022-03-11 20:02:31 [DEBUG]  		STANDARD ERROR:
2022-03-11 20:02:31 [DEBUG]  			/bin/bash: [running: command not found
2022-03-11 20:02:31 [DEBUG]  			/bin/bash: line 1: [running: command not found
2022-03-11 20:02:31 [DEBUG]  		SUCCESS RESPONSE CODE: 0, EXECUTION TIME: 3.82s
2022-03-11 20:02:31 [DEBUG]  		EXECUTING SYSTEM COMMAND: /bin/bash -c "[running ~/.bashrc]
[running common_alias.bash]
/usr/bin/python3 -m pip show aiortc"
2022-03-11 20:02:41 [DEBUG]  		STANDARD ERROR:
2022-03-11 20:02:41 [DEBUG]  			/bin/bash: [running: command not found
2022-03-11 20:02:41 [DEBUG]  			/bin/bash: line 1: [running: command not found
2022-03-11 20:02:41 [DEBUG]  			WARNING: Package(s) not found: aiortc
2022-03-11 20:02:41 [DEBUG]  		ERROR RESPONSE CODE: 1, EXECUTION TIME: 9.51s
2022-03-11 20:02:41 [DEBUG]  		EXECUTING SYSTEM COMMAND: /bin/bash -c "/usr/bin/python -c \"
import warnings
warnings.filterwarnings('ignore')
import json
import platform
import sys

try:
    import distro
except ImportError:
    # we don't have distro module
    distro = lambda: None
    distro.linux_distribution = lambda: 'Failed to get: Distro is not available.'

try:
    from pip import __version__ as pip_version
except ImportError:
    # we don't have pip module
    pip_version = '0.0.0 (Pip is not available.)'

try:
    from packaging import __version__ as packaging_version
except ImportError:
    # we don't have pip module
    packaging_version = '0.0.0 (Packaging is not available.)'

try:
    from packaging import tags
    pip_supported_tags = list(tags.cpython_tags())
except ImportError:
    try:
        if pip_version < '20.1' and pip_version >= '10.0':
            from pip._internal import pep425tags
            pip_supported_tags = pep425tags.get_supported()
        elif pip_version >= '1.4':
            from pip import pep425tags
            pip_supported_tags = pep425tags.get_supported()
        else:
            pip_supported_tags = []
    except ImportError:
        pip_supported_tags = []

if pip_supported_tags and hasattr(pip_supported_tags[0], 'platform'):
    pip_supported_tags = [(tag.interpreter, tag.abi, tag.platform) for tag in pip_supported_tags]

print(json.dumps({
    'platform_system': platform.system(),
    'platform_release': platform.release(),
    'platform_machine': platform.machine(),
    'platform_processor': platform.processor(),
    'platform_linux_distribution': 
        platform.linux_distribution() if hasattr(platform, 'linux_distribution') else distro.linux_distribution(),
    'platform_mac_ver': platform.mac_ver(),
    'platform_uname': platform.uname(),
    'platform_python_implementation': platform.python_implementation(),
    'platform_python_version': platform.python_version(),
    'platform_python_compiler': platform.python_compiler(),
    'platform_python_build': platform.python_build(),
    'pip_supported_tags': pip_supported_tags,
    'pip_version': pip_version,
    'packaging_version': packaging_version,
}))
\""
2022-03-11 20:02:44 [DEBUG]  		SUCCESS RESPONSE CODE: 0, EXECUTION TIME: 3.58s
2022-03-11 20:02:44 [CRITI]  [31mCould not upload installation report: <type 'exceptions.ValueError'> No JSON object could be decoded[39m
2022-03-11 20:02:44 [DEBUG]  Traceback (most recent call last):
  File "<stdin>", line 2188, in upload_report
  File "<stdin>", line 169, in wrapped
  File "<stdin>", line 984, in platform_details
  File "/usr/lib/python2.7/json/__init__.py", line 339, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python2.7/json/decoder.py", line 364, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python2.7/json/decoder.py", line 382, in raw_decode
    raise ValueError("No JSON object could be decoded")
ValueError: No JSON object could be decoded

2022-03-11 20:02:44 [CRITI]  [31m    COMPLETE[39m
2022-03-11 20:02:44 [CRITI]  [31m[39m
2022-03-11 20:02:44 [CRITI]  [31m################################################################################[39m
